src/Aspire.Hosting.Azure/IAzureKeyVaultSecretReference.cs

```diff
@@ -20,5 +20,10 @@ public interface IAzureKeyVaultSecretReference : IValueProvider, IManifestExpres
     /// </summary>
     IAzureKeyVaultResource Resource { get; }
 
-    IEnumerable<object> IValueWithReferences.References => [Resource];
+    /// <summary>
+    /// The resource that writes this secret to the Key Vault.
+    /// </summary>
+    public IResource? SecretOwner { get => null; set { } }
```

Comment:
Should this throw instead of being silent?

Reply:
I'll make it throw.